include/hip/hcc_detail/math_functions.h

```diff
@@ -1034,42 +1040,44 @@ double yn(int n, double x)
 }
 
 // BEGIN INTRINSICS
-// __DEVICE__
-// inline
-// double __dadd_rd(double x, double y) { return __ocml_add_rtn_f64(x, y); }
-// __DEVICE__
-// inline
-// double __dadd_rn(double x, double y) { return __ocml_add_rte_f64(x, y); }
-// __DEVICE__
-// inline
-// double __dadd_ru(double x, double y) { return __ocml_add_rtp_f64(x, y); }
-// __DEVICE__
-// inline
-// double __dadd_rz(double x, double y) { return __ocml_add_rtz_f64(x, y); }
-// __DEVICE__
-// inline
-// double __ddiv_rd(double x, double y) { return __ocml_div_rtn_f64(x, y); }
-// __DEVICE__
-// inline
-// double __ddiv_rn(double x, double y) { return __ocml_div_rte_f64(x, y); }
-// __DEVICE__
-// inline
-// double __ddiv_ru(double x, double y) { return __ocml_div_rtp_f64(x, y); }
-// __DEVICE__
-// inline
-// double __ddiv_rz(double x, double y) { return __ocml_div_rtz_f64(x, y); }
-// __DEVICE__
-// inline
-// double __dmul_rd(double x, double y) { return __ocml_mul_rtn_f64(x, y); }
-// __DEVICE__
-// inline
-// double __dmul_rn(double x, double y) { return __ocml_mul_rte_f64(x, y); }
-// __DEVICE__
-// inline
-// double __dmul_ru(double x, double y) { return __ocml_mul_rtp_f64(x, y); }
-// __DEVICE__
-// inline
-// double __dmul_rz(double x, double y) { return __ocml_mul_rtz_f64(x, y); }
+#if defined OCML_BASIC_ROUNDED_OPERATIONS
+__DEVICE__
+inline
+double __dadd_rd(double x, double y) { return __ocml_add_rtn_f64(x, y); }
+__DEVICE__
+inline
+double __dadd_rn(double x, double y) { return __ocml_add_rte_f64(x, y); }
+__DEVICE__
+inline
+double __dadd_ru(double x, double y) { return __ocml_add_rtp_f64(x, y); }
+__DEVICE__
+inline
+double __dadd_rz(double x, double y) { return __ocml_add_rtz_f64(x, y); }
+__DEVICE__
+inline
+double __ddiv_rd(double x, double y) { return __ocml_div_rtn_f64(x, y); }
+__DEVICE__
+inline
+double __ddiv_rn(double x, double y) { return __ocml_div_rte_f64(x, y); }
+__DEVICE__
+inline
+double __ddiv_ru(double x, double y) { return __ocml_div_rtp_f64(x, y); }
+__DEVICE__
+inline
+double __ddiv_rz(double x, double y) { return __ocml_div_rtz_f64(x, y); }
+__DEVICE__
+inline
+double __dmul_rd(double x, double y) { return __ocml_mul_rtn_f64(x, y); }
+__DEVICE__
+inline
+double __dmul_rn(double x, double y) { return __ocml_mul_rte_f64(x, y); }
+__DEVICE__
+inline
+double __dmul_ru(double x, double y) { return __ocml_mul_rtp_f64(x, y); }
+__DEVICE__
+inline
+double __dmul_rz(double x, double y) { return __ocml_mul_rtz_f64(x, y); }
+#endif
 __DEVICE__
```

Comment:
These rcp implementations are simply not correct..  They should be guarded too, and can use the same guard since they will use OCML for their implementation.

Reply:
I'll guard the rcp as well, but I won't change which ocml function it calls since it doesn't seem like ROCDL has __ocml_rcp_rt[n/p/z/e]_f[32/64] functions yet.